pkg/cmd/target.go

```diff
@@ -739,6 +769,70 @@ func gardenWrapper(targetReader TargetReader, targetWriter TargetWriter, configR
 	return nil
 }
 
+func serverWrapper(reader ConfigReader, serverName string) {
+	config := reader.ReadConfig(pathGardenConfig)
+
+	if findIP(serverName) != "" {
+		for _, garden := range config.GardenClusters {
+			kc := garden.KubeConfig
+			svr := getServerValueFromKubeconfig(kc)
+			addrIP, err := net.LookupIP(removeHTTPSandPort(svr))
```

Comment:
use url.ParseRequestURI to get the fields you are interested in, e.g.
```
u, err := url.ParseRequestURI(svr)
checkError(err)
addrIP, err := net.LookupIP(u.Host)
```

Reply:
> https://github.com/gardener/gardenctl/pull/180#discussion_r401579780
> https://github.com/gardener/gardenctl/pull/180#discussion_r401580188
> https://github.com/gardener/gardenctl/pull/180#discussion_r401601726

Hi @petersutter ,

For these three comments i'd like to have some discussion on this.
The core problem is to compare whether `gardenctl target server serverString` serverString matches server value in robot.kubeconfig.

I assume the server value in robot.kubeconfig is https://hostname:443 format, right? correct me if i was wrong.

My initial design for serverString are both https://hostname:443 and https://ipaddress:443.

So you would like to have following acceptable serverString:

1) hostname
2) ipaddress
3)hostname:port -> the port must be 443, am i right?
4)ipaddress:port
5) https://hostname:443
6) https://ipaddress:443
7) https://hostname
8) https://ipaddress

am i right? 

if above 8 scenario are all possible input then i will try to refactor my code to accept these 8 kinds of inputs.

Thanks.
-Neo

Reply:
I hope https://github.com/neo-liang-sap/gardenctl/pull/1  covers all these cases, PTAL